app/services/foreman_seeder.rb

```diff
@@ -34,6 +35,15 @@ def hash
   end
 
   def execute
+    Rails.logger.info("Aquiring Advisory-Lock 0x#{ADVISORY_LOCK.to_s(16)} for seeding")
+    ActiveRecord::Base.connection.select_value("SELECT pg_advisory_lock(#{ADVISORY_LOCK})")
```

Comment:
First process comes in, locks the lock, this return immediatelly. What happens when another process tries to lock it? Does it block?

Reply:
When using `pg_advisory_lock()`, it will block until Lock is released (either by explicit unlock or ending the DB-session).

Reply:
The commented code below this will just skip seeding if the lock is already taken.
Although this probably is not what we want :thinking: 

We could also implement a wait-loop ourselves, which skips if waiting for more than `n` seconds.

Reply:
Can it happen that the task, that acquired the lock, is interrupted and cannot release the resource?

Reply:
The `ensure`-block should is always called, even if the method is left by a raised exception.

Ultimatively, PostgreSQL advisory-locks are released when the DB-connection/-session ends.

Reply:
You can also create an explicit transaction to make sure the lock will be released at the end of the transaction (block).